### PR TITLE
Add a new utility function to test the user's Credentials. Use it in the 'init-creds' command.

### PR DESCRIPTION
**Pull request by BenBirt**

### CODE REVIEW

File: api/utils.ts [hunk 1]
```diff
@@ -49,6 +50,28 @@ export function coerceCredentials(warehouse: string, credentials: any): Credenti
   }
 }
 
+export async function testCredentials(credentials: Credentials, warehouse: string) {
```
Comment (lewish) [inline api/utils.ts]: if we are going to call this via the api should it not be in it's own command file like the other commands? Seems strange for us to import this from `@dataform/api/utils`

File: cli/index.ts [hunk 2]
```diff
@@ -173,10 +182,19 @@ const builtYargs = createYargsCli({
...
+          try {
+            await utils.testCredentials(credentials, argv.warehouse);
+            writeStdOut(commandOutput("Warehouse test query completed successfully."));
+          } catch (e) {
+            throw new Error(`Warehouse test query failed: ${e.stack || e.message}`);
+          }
+        }
+        const filePath = path.resolve(argv["project-dir"], CREDENTIALS_FILENAME);
+        fs.writeFileSync(filePath, prettyJsonStringify(credentials));
+        writeStdOut(commandOutput("Credentials file written:"));
```
Comment (lewish) [inline cli/index.ts]: Want to add a comment saying to change settings that you can edit this file directly?

File: api/utils.ts (hunk 1 above)
Comment (lewish) [reply api/utils.ts]: Either way, you want to export this in the api/index.ts file probably so it can just be used directly from importing `@dataform/api`

Comment (BenBirt) [reply api/utils.ts]: As discussed, moved all credentials related stuff to `credentials.ts` and exported that.

File: cli/index.ts (hunk 2 above)
Comment (BenBirt) [reply cli/index.ts]: Done

**Pull request merged**